### PR TITLE
[GEARPUMP-192] refactor example sources task to use DataSourceAPI

### PR DESCRIPTION
**Pull request by Roshanson**
[GEARPUMP-192]  Refactor example sources task to use DataSourceAPI.

### CODE REVIEW

**Comment by manuzhang** [comment]
please modify the commit comment to the format "[GEARPUMP-XXX] xyz"

**Comment by manuzhang** [comment]
also, please fix the UT failure

File: examples/streaming/wordcount/src/main/scala/org/apache/gearpump/streaming/examples/wordcount/Split.scala [hunk 1]
```diff
@@ -23,29 +23,34 @@ import java.util.concurrent.TimeUnit
...
+import org.apache.gearpump.streaming.source.{DataSource, DataSourceTask, Watermark}
 import org.apache.gearpump.streaming.task.{Task, TaskContext}
 
-class Split(taskContext: TaskContext, conf: UserConfig) extends Task(taskContext, conf) {
-  import taskContext.output
 
-  override def onStart(startTime: Instant): Unit = {
-    self ! Watermark(Instant.now)
-  }
+class Split() extends DataSource {
```
Comment (huafengw) [inline examples/streaming/wordcount/src/main/scala/org/apache/gearpump/streaming/examples/wordcount/Split.scala]: I prefer `class Split extends DataSource`, no braces.

File: examples/streaming/wordcount/src/main/scala/org/apache/gearpump/streaming/examples/wordcount/Split.scala [hunk 2]
```diff
@@ -23,29 +23,34 @@ import java.util.concurrent.TimeUnit
...
-    taskContext.scheduleOnce(Duration(100, TimeUnit.MILLISECONDS))(self !
-      Watermark(Instant.now))
+    Message("message")
   }
+
+  override def close(): Unit = {}
+
+  override def getWatermark: Instant = Instant.now()
+
+  Watermark(Instant.now)
```
Comment (huafengw) [inline examples/streaming/wordcount/src/main/scala/org/apache/gearpump/streaming/examples/wordcount/Split.scala]: What's this line of code used for?

Comment (Roshanson) [inline examples/streaming/wordcount/src/main/scala/org/apache/gearpump/streaming/examples/wordcount/Split.scala]: Thanks，i'm learning how  gearpump  is working now.

File: examples/streaming/wordcount/src/main/scala/org/apache/gearpump/streaming/examples/wordcount/Split.scala [hunk 3]
```diff
@@ -19,34 +19,45 @@
...
+import scala.collection.mutable.ArrayBuffer
 
-class Split() extends DataSource {
 
+class Split extends DataSource {
+
+  val result = ArrayBuffer[Message]()
+  var item = -1
+  Split.TEXT_TO_SPLIT.lines.foreach { line =>
+    line.split("[\\s]+").filter(_.nonEmpty).foreach { msg => //  => 为匿名函数,传入一个msg参数执行右边的操作
```
Comment (manuzhang) [inline examples/streaming/wordcount/src/main/scala/org/apache/gearpump/streaming/examples/wordcount/Split.scala]: please remove this comment.

**Comment by manuzhang** [comment]
+1

**Pull request closed without merging**